Duplicati/Library/Main/Operation/Backup/BackendUploader.cs

```diff
@@ -318,7 +318,15 @@ private async Task<bool> DoWithRetry(Func<Task> method, FileEntryItem item, Work
             for (retryCount = 0; retryCount <= m_options.NumberOfRetries; retryCount++)
             {
                 if (m_options.RetryDelay.Ticks != 0 && retryCount != 0)
-                    await Task.Delay(m_options.RetryDelay).ConfigureAwait(false);
+                {
+                    TimeSpan delay;
+                    if (m_options.RetryDelayExponential)
+                        delay = TimeSpan.FromTicks(m_options.RetryDelay.Ticks * (2L << Math.Min(63, retryCount - 1)));
+                    else
+                        delay = m_options.RetryDelay;

```

Comment:
To avoid duplicating the logic in `BackendManager`, could we extract this to a method in the `Duplicati.Library.Utility.Utility` class?  Something like
```
public static TimeSpan GetRetryDelay(TimeSpan retryDelay, int retryAttempt, bool useExponentialBackoff)
{
    if (retryAttempt < 0)
    {
        // throw exception
    }
    ...
}
```
This would also make it easier to add a really simple unit test to the `Duplicati.UnitTest.UtilityTest` class to verify the behavior.

Reply:
Done. I used your template code to check the `retryAttempt` parameter. Not being familiar with the project conventions, I'm not sure whether the other parameters should be checked as well.

I also added the test.